From e54f65df2bcb68b4493832c83c141da9f579964e Mon Sep 17 00:00:00 2001
From: Remi Collet <fedora@famillecollet.com>
Date: Fri, 24 Jul 2015 09:57:36 +0200
Subject: [PATCH] Fix: json_post.flags=2 ignored with php7

Detected on 32bits build, where 003.phpt is failing
(not detected on 64bits, because not covered)
---
 package.xml     | 1 +
 php_json_post.c | 2 +-
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/php_json_post.c b/php_json_post.c
index fd3165c..4a0fcd3 100644
--- a/php_json_post.c
+++ b/php_json_post.c
@@ -63,7 +63,7 @@ static SAPI_POST_HANDLER_FUNC(php_json_post_handler)
 
 			ZVAL_NULL(&tmp);
 
-			php_json_decode(&tmp, json->val, json->len, JSON_POST_G(flags), PG(max_input_nesting_level));
+			php_json_decode_ex(&tmp, json->val, json->len, JSON_POST_G(flags), PG(max_input_nesting_level));
 
 			switch (Z_TYPE(tmp)) {
 			case IS_OBJECT:
